python/ray/_private/log.py

```diff
@@ -27,6 +28,21 @@ def clear_logger(logger: Union[str, logging.Logger]):
     logger.handlers.clear()
 
 
+class RayDataInfoFilter(logging.Filter):
+    """Filters out info and debug messages from Ray Data.
+
+    This prevents Ray Data from spamming users with info and debug messages.
```

Comment:
No, because if we set the log level to `logging.WARN`, then we won't log info messages to the `ray-data.log` file.

Reply:
Rather than filtering messages from Ray Data, an alternative implementations is to not propagate messages to the `"ray"` logger.

Here are the tradeoffs I can think of:

**Advantages of propagating**
* Can leverage existing configuration for logging to console.

**Disadvantages of propagating**
* Broken abstraction barriers. General Ray logging configuration has Ray Data-specific filter.

**Advantages of not propagating**
* Don't need to add Ray Data-specific filter to general Ray logging configuration

**Disadvantages of not propagating**

- Need to configure logging to console at Data-level. Options are to:
   - (1) Duplicate exact configuration.
   - (2) Import configuration, and couple `ray.data._internal.logging` with `ray._private.log`.
   - (3) Have separate but inconsistent configurations.

Looks like Serve chose to not propagate messages to the top-level Ray logger.

Reply:
Refactored to remove `RayDataInfoFilter`